app/src/Components/Canvas/SceneManager.js

```diff
@@ -0,0 +1,113 @@
+import * as THREE from 'three';
+import RectangleView from "./RectangleView";
+
+export default class SceneManager {
+    constructor(center, screenDimentions) {
+        const { x, y } = center;
+        const { width, height } = screenDimentions;
+
+        this.scale = 1;
+        this.aspectRatio = 4;
+
+        this.scene = new THREE.Scene();
+        this.viewRect = new RectangleView(x, y,
+            width / this.scale * this.aspectRatio,
+            height / this.scale * this.aspectRatio);
+        this.objects = [];
+    }
+
+    _drawObjects() {
+        this.objects.forEach(object => {
+            // if (object.inRectangle(this.viewRect)) {
```

Comment:
This is impossible! After checking your code I saw that these objects are objects from THREE.js. They haven't an `Element` interface.

Reply:
Fixed

Reply:
No

Reply:
YES!